### PR TITLE
String/exercise 04

### PR DESCRIPTION
**Pull request by ntnghia0320**

### CODE REVIEW

File: src/main/java/StringExercise04.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+public class StringExercise04 {
+    static void computeLPSArray(String str, int M, int[] lps) {
+        int len = 0;
+        int i;
+
+        lps[0] = 0;
+        i = 1;
```
Comment (voquanghoa) [inline src/main/java/StringExercise04.java]:
```suggestion
        lps[0] = 0;
        int i = 1;
```

File: src/main/java/StringExercise04.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+public class StringExercise04 {
+    static void computeLPSArray(String str, int M, int[] lps) {
+        int len = 0;
```
Comment (voquanghoa) [inline src/main/java/StringExercise04.java]:
```suggestion
        int repeatLength = 0;
```

File: src/main/java/StringExercise04.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+public class StringExercise04 {
+    static void computeLPSArray(String str, int M, int[] lps) {
```
Comment (voquanghoa) [inline src/main/java/StringExercise04.java]:
```suggestion
    static void computeLPSArray(String str, int[] lps) {
```

File: src/main/java/StringExercise04.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        int len = 0;
+        int i;
+
+        lps[0] = 0;
+        i = 1;
+
+        while (i < M) {
```
Comment (voquanghoa) [inline src/main/java/StringExercise04.java]:
```suggestion
        while (i < str.length) {
```

File: src/main/java/StringExercise04.java (hunk 3 above)
Comment (voquanghoa) [inline src/main/java/StringExercise04.java]:
```suggestion
    public boolean isRepeatString(String str)
```

Comment (ntnghia0320) [reply src/main/java/StringExercise04.java]: Done

File: src/main/java/StringExercise04.java (hunk 4 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise04.java]: Done

File: src/main/java/StringExercise04.java (hunk 3 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise04.java]: Done

File: src/main/java/StringExercise04.java (hunk 2 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise04.java]: Done

File: src/main/java/StringExercise04.java (hunk 1 above)
Comment (ntnghia0320) [reply src/main/java/StringExercise04.java]: Done

**Pull request merged**